bundles/io/org.eclipse.smarthome.io.console/src/main/java/org/eclipse/smarthome/io/console/internal/extension/ItemConsoleCommandExtension.java

```diff
@@ -78,6 +99,42 @@ public void execute(String[] args, Console console) {
         }
     }
 
+    private void removeTag(String itemName, String tag, Console console) {
+        Item item = itemRegistry.get(itemName);
+        if (item instanceof GenericItem) {
+            GenericItem gItem = (GenericItem) item;
+            gItem.removeTag(tag);
+            Item oldItem = itemRegistry.update(gItem);
+            if (oldItem == null) {
+                console.println("Error: Cannot remove tag " + tag + " from item " + itemName
+                        + " because this item does not belong to a ManagedProvider");
+            } else {
+                console.println("Successfully removed tag " + tag + " from item " + itemName);
+            }
+        } else {
+            console.println(
+                    "Error: Cannot remove tag " + tag + " from item " + itemName + " because it is not a GenericItem");
+        }
+    }
+
+    private void addTag(String itemName, String tag, Console console) {
+        Item item = itemRegistry.get(itemName);
+        if (item instanceof GenericItem) {
+            GenericItem gItem = (GenericItem) item;
+            gItem.addTag(tag);
```

Comment:
If I want to pass `GenericItem::removeTag` and `GenericItem::addTag`to a method that would handle adding and removing of tags the compiler complains about that `GenericItem` is not a functional interface, which is correct since it has more than one method. Thats why I left it as 2 methods for now, but changed the order. 

But maybe I completely misunderstood how how you wanted to refactor the code to use method references... :confused: